lib/ansible/modules/identity/keycloak/keycloak_clienttemplate.py

```diff
@@ -311,7 +364,20 @@ def main():
     updated_clientt.update(changeset)
 
     result['proposed'] = changeset
-    result['existing'] = before_clientt
+    if state == 'dump':
+        result['msg'] = 'Client template %s has been dumped into "existing" variable.' % result['existing']['name']
+        module.exit_json(**result)
+
+    # partially validate protocolmappers
+    if 'protocolMappers' in changeset:
```

Comment:
You should be able to handle this in the argument_spec, as described at http://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-block :

> suboptions:  If this option takes a dict, you can define it here. See azure_rm_securitygroup, os_ironic_node for examples

Reply:
Thank you for that pointer. That validation is now done via argument_spec and documented with suboptions.